Sources/Overload/OvTools/src/OvTools/Utils/SystemCalls.cpp

```diff
@@ -30,3 +31,25 @@ void OvTools::Utils::SystemCalls::OpenURL(const std::string& p_url)
 {
 	ShellExecute(0, 0, p_url.c_str(), 0, 0, SW_SHOW);
 }
+
+std::string OvTools::Utils::SystemCalls::GetPathToAppdata()
+{
+	PWSTR path = nullptr;
+	std::string result;
+
+	HRESULT getPath = SHGetKnownFolderPath(FOLDERID_RoamingAppData, 0, nullptr, &path);
+	
+	// We get a wide string back which we need to convert back to a utf-8 string
+	if (SUCCEEDED(getPath)) {
+		// If size == 0, we have errors and are not able to convert the wide string
+		int size = WideCharToMultiByte(CP_UTF8, 0, path, -1, nullptr, 0, nullptr, nullptr);
+		if (size > 0) {
+			std::string converted(size - 1, 0);
+			WideCharToMultiByte(CP_UTF8, 0, path, -1, &converted[0], size, nullptr, nullptr);
+			result = converted;
+		}
+		CoTaskMemFree(path);
+	}
+	
+	return result;
```

Comment:
Here is a more modern C++ approach, RAII-compliant, with some additional `const` qualifiers.

```suggestion
    PWSTR rawPath = nullptr;
    const HRESULT hr = SHGetKnownFolderPath(FOLDERID_RoamingAppData, 0, nullptr, &rawPath);
    std::unique_ptr<wchar_t, decltype(&CoTaskMemFree)> path(rawPath, CoTaskMemFree);
    
    OVASSERT(SUCCEEDED(hr), "Failed to get AppData path");
    
    const int size_needed = WideCharToMultiByte(CP_UTF8, 0, path.get(), -1, nullptr, 0, nullptr, nullptr);
    std::string appDataPath(size_needed, 0);
    WideCharToMultiByte(CP_UTF8, 0, path.get(), -1, &appDataPath[0], size_needed, nullptr, nullptr);
    
    return appDataPath;
```